### ISSUE TITLE
Any news about udp port failed to open ?

### ISSUE THREAD

**Issue by Ready4Next** [moonlight-stream/moonlight-android#230]
Hi,

I have the same problem as #217, and I guess that it is corrected somewhere in moonlight-common (see moonlight-embedded).

My specs are (if it matters anyway):

> Intel 6700K
> GF GTX980 Ti Xtreme Windforce 6Gb
> 32Gb DDR4@3200
> GFExperience v2.11.3.5

Android System 1: 

> Minix X8-H Plus

Android System 2: 

> Galaxy S5 Neo

Network:

> WiFI 5Ghz for Samsung
> LAN 1Gbps for Minix & main computer

Same result when testing steam games or desktop streaming on both android machines. I will be happy to help if needed :).

Thanks

**Comment by cgutman** [comment]
Do you have any firewall or AV software on your PC?

**Comment by ShawNMusliM** [comment]
@Ready4Next I have having the same problem. Are you getting rtsp handshake error?

When I open the ports that need to be opened, both TCP and UDP, my TCP ports open fine. Unfortunately, my UDP ports, when I use port checker, never actually open it seems. I use the exact same process for opening the UDP ports that I use for opening the TCP ports but when I run port checker on port 47998 for example it is closed or times out. Ive been battling this issue all day and its killing me. I just dont understand why the UDP ports will not open -.-' No firewall enabled through windows, no AV program even installed... the UDP ports just wont fucking open and its killing me inside.

**Comment by ShawNMusliM** [comment]
https://forums.geforce.com/default/topic/826090/remote-gamestream-broken-again-ports-changed-again-/?offset=13

If you go down to post #11 that is the exact problem I am having it seems. Ive opened every port mentioned in your setup guide and every port on that thread and still all I can do is see my games and launch them. When its time for the game image to appear I still get "Connection Error, Failed to start RTSP handshake ok, could not send Enet package?(packet?)"

And still, after having opened those UDP ports, and many more than the ones you list on the setup page, my port checker site shows them ALL to be closed.

**Comment by Ready4Next** [comment]
Hi,

All I have on my desktop is the windows firewall, and I have already tried with it totally disabled. All the ports (both TCP & UDP) are configured to be opened by nvstream, even on my router. I will make a list of them when I'm home.

BUT, I don't have checked by myself if the ports CAN truly be opened, thanks for your suggestion @ShawNMusliM, I will get back to this thread with my results :)

**Comment by ShawNMusliM** [comment]
@ready4next how have you configured nvstream to interact with your ports? That is, what did you configure in nvstream and where did you configure?

**Comment by ShawNMusliM** [comment]
Yea, I just turned windows firewall off completly and it didn't help. It's either something in the code or our ISP's do not want those ports opened. Or we may have to request them to open those ports. I'm hoping the developer here gets back with us soon. 
On May 20, 2016 10:51 AM, Junta <notifications@github.com> wrote:@ShawNMusliM It don't work either... But what I did is let nvstreamer, NvStreamNetworkService & NvStreamUserAgent the ability top open any UDP if needed in windows firewall (network properties | firewall | advanced properties) both ingoing and outgoing traffic.

The result is not really convincing for UDP:

C:\WINDOWS\system32>netstat -anb | find "48010"
  TCP    0.0.0.0:48010          0.0.0.0:0              LISTENING


and for portqry:

C:\WINDOWS\system32>PortQry.exe -n 192.168.0.23 -p both -e 48010

Querying target system called:

 192.168.0.23

Attempting to resolve IP address to a name...


IP address resolved to xxx.be

querying...

TCP port 48010 (unknown service): LISTENING

UDP port 48010 (unknown service): NOT LISTENING


But maybe I'm doing something wrong/forgot something important ?
Arr... Yes, I'm using Windows 10 (forgot to tell, sorry)...
Any other ideas ?

**Comment by Ready4Next** [comment]
Hi,

Here is a wireshark capture, but I don't know where to look at. All I see is a lot of [RANGE_ERROR] [BLOCK_DROPPED] [PACKET_RESEND] between my desktop and my phone's local IP address, with the protocol GVSP. I even don't know what the meaning of what I'm writing :).

Anyway, here's a link for the pcap file for those who can read it and maybe help me: http://workupload.com/file/Td6CBt6

Many thanks in advance :)

**Comment by cgutman** [comment]
@Ready4Next It looks like the PC is responding to the connect request but the phone isn't acknowledging the connect response. Can you tell me what error you saw when the connection failed? Was it "Failed to receive ENet packet" or "Unable to connect to UDP port 48010"?

**Comment by Ready4Next** [comment]
I have "Unable to connect to UDP port 48010" on both (my phone and the minix android box)... There are both android 5 (if you need the exact version, please tell me).

What can I do for you to help solving this ?

Thanks a lot

**Comment by cgutman** [comment]
The Wireshark capture and the Moonlight error message are very helpful to tell me what function is failing. Just to complete the report, can you upload a zip of your C:\ProgramData\NVIDIA Corporation\NvStream directory? ProgramData is a hidden folder so you may need to enable hidden files to see it.

I'm going to try to reproduce the problem using a protocol debugging utility and Wireshark with my setup.

**Comment by Badhoagie91** [comment]
My phone boots steam on my PC, but then the phone gets a connection error "failed to start RTSP handshake" and then "Failed to receive ENET Packet"

**Comment by Ready4Next** [comment]
Here are all NV* folders containing logs in them, in case NvStream folder alone isn't enough :)

http://workupload.com/file/rk4SXKP

Thanks

**Comment by cgutman** [comment]
@Ready4Next thanks, one more thing. Can you get ADB logcat logs of you opening Moonlight, then trying to stream, and upload those too?

**Comment by Ready4Next** [comment]
@cgutman As you requested, here's a 2Mb+ log from adb logcat captured from my phone (I do not have filtered anything), compressed in rar format.

http://workupload.com/file/WPrg3up

Thanks :)

**Comment by cgutman** [comment]
@Ready4Next well, good news and bad news. Good news is that I found a longstanding bug that can cause RTSP handshake to fail over the Internet. I believe this to be one of the major causes of the "Failed to receive ENet packet" message. Bad news is I wasn't able to reproduce your specifc issue.

@ShawNMusliM @Badhoagie91 Can you try this build and see if you still get the error? https://onedrive.live.com/redir?resid=C15BF9770619F56!162078&authkey=!AKqCYj_nD7pbvaM&ithint=file%2capk

**Comment by ShawNMusliM** [comment]
@cgutman Downloading it now. Will let you know. BTW, it gave a new error recently. Said, could not connect to UDP port.

**Comment by ShawNMusliM** [comment]
@cgutman Ok, downloaded the new APK. Getting the same error man. 

Connection Error, Failed to start RTSP handshake, could not connect to UDP port 48010

**Comment by cgutman** [comment]
How are you connecting? LTE, your local network, or another network?

**Comment by ShawNMusliM** [comment]
4G LTE. Connecting through my own network, via wifi, works flawlessly. Some stuttering but it works.

**Comment by cgutman** [comment]
Can you try rebooting your PC? If that doesn't work, try restarting your router. We need to get back to that old error.

**Comment by ShawNMusliM** [comment]
Ok sec.

**Comment by cgutman** [comment]
@Ready4Next Please send me another ADB logcat log of you attempting to stream with this new build: https://onedrive.live.com/redir?resid=C15BF9770619F56!162079&authkey=!AF8Xq_Wtb86cj3I&ithint=file%2capk

**Comment by Ready4Next** [comment]
Hi all,

Happy to know that a bug has been vanished :) I'm wondering if it has anything to do with a synchronisation between the devices, a thing like phone: 'I need a port, open it, desktop: 'I opened port for you', but between both, the phone already send another request because he doesn't have received the response in time. After that, the desktop refuses the connection because it is already open. But maybe I'm wrong, I'm not skilled enough in network programming :).

Anyway, @cgutman, here is your logcat with the new APK :)

http://workupload.com/file/Sb4N3RT

Thanks

**Comment by Ready4Next** [comment]
I will try to increase the time out in RtspConnection from 10 sec to another value, just to see if it throw the same exception. The weird thing is that I can start and stop a game, but even if a game is launched and all is running on my desktop, both android machines hangs on starting streaming with rtsp error...

**Comment by Ready4Next** [comment]
Increased timeout to 30 sec, battle against idea to have the build dir of the class files => no luck, trying eclipse => I finally have limelight-common.jar file (grr I hate java, why not keep it simple after more than 2 decades of existence ?), launch android studio with new jar file and debug moonlight-android to my phone => ok.

Result: no luck, same error :-/.

**Comment by cgutman** [comment]
@Ready4Next 

I have a theory that something on your PC or network is preventing outbound UDP traffic from port 48010 from your PC. Do you have another PC that you can test with the Java client (moonlight-pc)?

I would be interested to see if you did a setup where you ran Wireshark captures on both PCs while trying to stream using Moonlight PC, then we can compare both captures to see if they both see the same packets.

**Comment by Ready4Next** [comment]
I'm sorry to say that it's my only desktop with a nvidia graphic card, the others machines are with amd or intel graphics...

What if I make a virtual machine (either VMWare or VirtualBox) ? Will it help in some ways ?

**Comment by cgutman** [comment]
Your other computer (the one running Moonlight PC) doesn't need an Nvidia graphics card.

**Comment by Ready4Next** [comment]
Ok then, I will make a complete setup out of my laptop, it will take some times though.

Is there any steps (FAQ) to make all work, do I need something in particular except wireshark ?

**Comment by cgutman** [comment]
Nope, it should just work. Moonlight PC should find your GeForce PC just like the other clients do. If not, you can type in the IP address of your PC manually. Click the pair button, go through the pairing process, then show app list. Start Wireshark capturing on both PCs, then try to start streaming. Stop Wireshark capture, save, and upload both captures.

**Comment by Ready4Next** [comment]
Ok, I have done it, but I'm using WIFI connection on laptop and captured all with wireshark, that makes a big file.

Here are the logs: http://workupload.com/file/FXnkJ2H

Sincerely, thank you for your patience :)

**Comment by Ready4Next** [comment]
Oh, and the result is "Unexpected error" :)

**Comment by cgutman** [comment]
Weird, that one seems to have got through RTSP handshake successfully. It looks like you've got a bunch of virtual network cards on your PC. Can you try disabling all network connections other than the Ethernet connection to your router, reboot your PC, and try streaming to Android again.

**Comment by Ready4Next** [comment]
I agree, weird problem, and it seems I'm part of a little percentage of users (if not the only one) that have this problem.

I tried what you suggested, disabled all vmware/wifi network connections to keep only ethernet and npcap connections enabled. I even disabled "Killer network manager" (the app that comes with my Gigabyte GA-Z97X-Gaming G1 motherboard), but no luck...

Anything you think of that I can try ?

**Comment by ShawNMusliM** [comment]
So no luck with this problem? I dont see why something on my desktop would be preventing this port from opening. Like I said, my windows firewall is turned off and I have zero antivirus programs installed. Furthermore, ive tried enabling windows firewall and anything that is connected to nvidia streaming I made exceptions to open all ports.

**Comment by cgutman** [comment]
@ShawNMusliM if it's working within the same network, it has to be something wrong with your router, port forwarding configuration, or ISP (unlikely). It could also be blocked on the other network that you're trying to stream from.

**Comment by ShawNMusliM** [comment]
Ive changed my router out twice now and the proccess to open my UDP ports is the same as opening the TCP ports. And two of my TCP ports open just fine. When my TCP ports are not open I cant see my computer or my game list at all. It could very well be my ISP or phone carrier doing this.

Im rooted if that makes any difference. The new APK build you asked me to try was for the normal moonlight app.

**Comment by cgutman** [comment]
@Ready4Next can you try streaming again with Moonlight PC (same as before) then send me another ZIP of your NVIDIA logs? That error indicates that the stream actually gets established then terminated (unlike what is happening on Android).

**Comment by cgutman** [comment]
@ShawNMusliM can you post a screenshot of your router's port forwarding configuration?

**Comment by ShawNMusliM** [comment]
Yea, sec.

**Comment by ShawNMusliM** [comment]
https://www.dropbox.com/s/wcw9v7sm790hckd/Screenshot%202016-05-22%2012.29.19.png?dl=0

**Comment by ShawNMusliM** [comment]
@cgutman On the phone right now with my ISP. They dont know of any thing that ATT does to block ports. And the process that he is going through to open UDP ports he is saying I did correctly.

**Comment by ShawNMusliM** [comment]
https://www.dropbox.com/s/z8esp9maz0qf1ic/Screenshot%202016-05-22%2012.43.03.png?dl=0

**Comment by cgutman** [comment]
@ShawNMusliM looks like UDP 48010 isn't in your port forwarding configuration, TCP 48010 is there but that's not used on modern GFE versions.

**Comment by Ready4Next** [comment]
@cgutman Here are the logs: http://workupload.com/file/BuH7cNs

**Comment by Ready4Next** [comment]
@cgutman while capturing those logs, moonlight-pc crashes without notice 2 or 3 times, here's a second log with no "mysterious" crashes and the same msgbox as result:http://workupload.com/file/Z9LakS9

**Comment by cgutman** [comment]
@Ready4Next I see this message in your log "Failed to create FBC. Another instance exists and concurrent capture is not supported." Can you check that you don't have any known incompatible programs on your system: https://github.com/moonlight-stream/moonlight-android/wiki/Troubleshooting#known-application-compatibility-issues

If not, can you think of any programs that may want to capture/record the screen that you may have installed? ShadowPlay can sometimes do this, so I'd disable it if you have it turned on.

@ShawNMusliM Can you try adding UDP 48010 to your port forwarding configuration and see if that resolves it? If you get back to the "Failed to receive ENet packet" error, try the build I gave you yesterday.

**Comment by ShawNMusliM** [comment]
Holy crap -.- I feel like such an idiot man. Im so sorry. Yea, I forgot to add that one UDP port and it went back to Enet error. Then I tried your new build and it connected fine.

@cgutman Really sorry, total waste of your time. Also, got my ONG usb and the xbox controls work flawless on supported games by nvidia.

Thanks again.

**Comment by cgutman** [comment]
@ShawNMusliM No worries, it happens. Your original bug was found and fixed, so it wasn't all for nothing.

**Comment by Ready4Next** [comment]
@cgutman Here are the new logs: http://workupload.com/file/kVd2exh

You were right for moonlight-pc, kino server was blocking the streaming, I can now get in game (just hear it in fact, because I have no video on the client, but the gamepad is responding well).

Concerning the android boxes, still the same ol'error ^^. So with the logs, you have an .old which contains semi-successful connections with moonlight-pc, and .curr wich contains same rtsp error with my phone...

Thanks for pointing this :)

**Comment by cgutman** [comment]
@Ready4Next On your other PC, can you run the v0.6.7 jar in a command-line window, try to stream, and copy the log output from that window?

You'll run either: java -jar [moonlight jar file]
or have to provide a full path like: C:\Program Files\Java\[some directory]\bin\java.exe -jar [moonlight jar file]

**Comment by Ready4Next** [comment]
Hi @cgutman 

Here is the logs: http://workupload.com/file/2HUmcnq

First I have tried to launch with default settings (720p60fps), ith throws an exception at ENet pump message, second I have tried to launch at 720p30fps as stated in the FAQ, it throws the same exception. It also appears that audio is too slow. 

I have tried to run from the source, but I'm stuck in LibraryHelper, when he tries to load a 'platform' file from multiple directories. If I force to 'win64' it hangs later in LoadNativeLibraries when loading libs from inexistant 'binlibs' dir...

I have also taken a third android device (a galaxy tab with android 4), same 'UDP connection failed at port 48010. I will do a search for which software can block this specific port, because I have checked all the other ports opened by Nv* files and nothing is wrong (at least from what I know).

Thanks again for your precious help :-)

**Comment by cgutman** [comment]
Can you describe what was happening before the stream terminated? That exception is normal when the stream is ended by the PC.

**Comment by Ready4Next** [comment]
It connects to the desktop, the desktop starts NVidia game stream, launch steam if necessary and launch the game.

Meanwhile, the client computer shows a progress, start RTSP shake, and start streaming (all connections seems to go well). No video is shown but I have sound. If I exit the game normally, then it crashes on client computer.

On aall the other devices, it launch the games too, but do not stream anything, just throw the same error on all three devices "UDP Connection failed at port 48010".

**Comment by cgutman** [comment]
There is very strange behavior in those packet captures. Moonlight PC only sends 1 UDP packet on port 47998 during your stream but it's supposed to send one every 500 ms, yet no errors are reported. Something strange is happening if you have sound, because I don't see any packets on UDP 48000 in either end's packet capture and that's the port that carries audio data. Is there some other connection like a VPN or something that the computers could be communicating over? Is there a firewall on your client PC that could be causing problems.

Nothing makes any sense at this point. I'm thinking it must be something in your network topology or software installed on your PCs interfering. If both PCs have gigabit NICs, you could try simply connecting them together with an Ethernet cable and attempting to stream that way. If that still doesn't work, we could start looking one by one at your installed programs, but honestly it's probably easier to just reinstall Windows. If you have a spare drive and some time, you could try installing a fresh copy of Windows on that drive and installing the bare essentials to get GameStream running (Steam, GeForce Experience, GeForce drivers) and see if it works there.

**Comment by Ready4Next** [comment]
Thanks for your support, but before whiping my development computer, I think I will try soft solutions first :). I will post my findings if it can help someone else.

Network in general and Java hates me :).

Regarding your questions: The laptop I used as client has the same environment as the desktop, it uses windows 10, windows firewall to default, standard microsoft development environment, SQL Server. Remote desktop is activated on both, but not used when testing. The desktop has three hardware network interface (2 LANs 1 WIFI) and common vmware network interface. They both have bluetooth. Each not relevant interface were disabled when testing. 

In my network topology, nothing fancy, a router connected to internet that has 1 direct connection to the Minix box, one internet access, one connection to Devolo dLAN 1200+ (cpl) and distribute WIFI on both 2.4Gh and 5 Ghz. The desktop server is connected to another devolo and linked to the local network on one LAN interface. The laptop client has been tested by wifi and direct link to devolo too, same result.  All machines are using DHCP with common IP Address class for a home network (192.168.0.x).

I have also tried a fourth android device (Yeah I'm totally crazy ^^) which is a rooted device (all the others aren't rooted), but no luck.

Just for my mental health, I will reinstall GForce again, create another small network with an old router I have somewhere in the attic, try to remove software one by one and test carefully.  I'm taking bets on which solution will be the right one :).

Thanks again for your help. Oh, by the way, did you pushed a working source version of moonlight-pc ?

**Comment by cgutman** [comment]
Moonlight-PC should work as long you specify the correct native library path in Eclipse. In the Moonlight project settings in Eclipse, in the library path tab, you should be able to specify a native library path. This works for me and is how I test when I'm developing.

Alternatively, you can just install ant and build jars by running ant in the root directory of the checkout.

**Comment by cgutman** [comment]
@Ready4Next I looked again at your logs from Moonlight PC. It looks like the OpenGL renderer may be breaking and that's why you don't see video. It's still a mystery where those packets are coming from though, since they don't appear in the pcap. Try streaming again with v0.6.7 instead of v0.7.5.

**Comment by Ready4Next** [comment]
@cgutman 
I have a good and bad news too :)

The good is that I'm able to stream and play from a shiny 5 inch screen instead of a 24 one ! Great, isn't it ? Yeah, you read it right, I have found something that goes wrong with my network. It seems that Devolo CPL kit that I have plugged between the router and me, and that is supposed to be "totally transparent" and let all data pass-through it as it didn't even exists is a totally wrong assumption. It let some packets goes through, but apparently not all are correctly redirected, and there's no way to configure it. Even if you clone WIFI from your router using WPS.

How did I do ? That's the tricky part. If the WIFI is cloned from existing connection AND the phone connects to your real router instead of that cloned WIFI router, it won't work. But if you are lucky enough, and it connects to the cloned one, it will work flawlessly. So I have made a new WIFI accesspoint from the CPL and forced the phone to connect through it for testing, and it works. Next will be to see why it acts like that and ask devolo for support :) as I don't know if it is a bug in their firmware or if it is just the "way it works". So within this thread 1 bug has been found, and 1 incompatibility issue may be added in the FAQ (I need more testing to be sure).

Regarding your questions:

> Did you specify the address to Moonlight PC using a DNS name, IP address, or using the autodiscovered PC list? Similarly for Android, did it find your PC automatically or did you add it manually?

=> It depends, on some devices, it is found auto-magically, and sometimes I need to force the IP address (I rarely use the name to access to a computer in my network, very old habit).

If you want to know why it don't works in my laptop, we can test it out, but don't worry too much on that as I will surely never use it to stream a game :) The objective is to stream to my android box on my TV. And as I move this box from my room to my living-room I need it to work in all my house. I will get there one day :).

Again thanks a lot for your patience (for reading an approximate english ^^), reactivity and support.

Have a nice day !
